### ISSUE TITLE
cmd/compile: inline small static initializers

### ISSUE THREAD

**Issue by randall77** [golang/go#18872]
```
type T struct {
	a, b, c, d int
}
func f(p *T) {
	*p = T{1, 2, 3, 4}
}
```
The code for `f` is:
```
(tmp1.go:8)	MOVQ	"".statictmp_0(SB), AX
(tmp1.go:8)	MOVQ	"".statictmp_0+8(SB), CX
(tmp1.go:8)	MOVQ	"".statictmp_0+16(SB), DX
(tmp1.go:8)	MOVQ	"".statictmp_0+24(SB), BX
(tmp1.go:8)	MOVQ	"".p+8(FP), SI
(tmp1.go:8)	MOVQ	AX, (SI)
(tmp1.go:8)	MOVQ	CX, 8(SI)
(tmp1.go:8)	MOVQ	DX, 16(SI)
(tmp1.go:8)	MOVQ	BX, 24(SI)
```
That's kind of dumb to copy from a statictmp.  Just use `MOVQ $1, (SI)` and so forth.

**Comment by josharian** [comment]
This was broken by the fix to #12841. The trick here is to figure out how to selectively disable that optimization so that #12841 doesn't regress.

**Comment by josharian** [comment]
CL shortly.
